Automattic/wp-calypso
# Woo Onboarding: Add email verification functionality

#### Proposed Changes

* Clicking on the "Resend email verification" button should send a verification email.
* If the email has already been verified or if there's another failure, an error message should be displayed.
* The button text should change to provide user feedback.

**Visuals**

https://user-images.githubusercontent.com/2124984/172865663-24e96d9d-70b4-46d9-9413-292a96808bcf.mov

<img width="748" alt="Screen Shot 2022-06-09 at 10 15 48 AM" src="https://user-images.githubusercontent.com/2124984/172868871-28dd742e-7a01-41b5-9c2d-d56c8359370b.png">


#### Testing Instructions

* Switch to this PR, navigate to `/setup/wooVerifyEmail?siteSlug=[your site slug]&flags=signup/woo-verify-email`
* Test both with an account that has not yet been verified and one that has.
* If the email needs verification, the button should display "Sending..." then "Request sent!"
* If the email can't be verified, an error message should be shown.

Fixes #64163 #64461
